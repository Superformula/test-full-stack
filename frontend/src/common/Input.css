.input-label {
    font-weight: 600;
    font-size: 18px;
    line-height: 23px;
    margin-top: 0;
}

.input-container {
    margin: 25px 0;
}

.input-text-box {
    width: 98%;
    height: 40px;
    background: #FFFFFF;
    border: 1px solid rgba(0, 0, 0, 0.1);
    border-radius: 8px;
    padding-left: 10px;
}
