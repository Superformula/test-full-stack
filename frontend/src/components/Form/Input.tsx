import { forwardRef } from "react";
import styles from "./Input.module.css";

interface Props {
  name: string;
  defaultValue: string;
}

const Input = forwardRef<HTMLInputElement, Props>((props: Props, ref) => {
  const { name, defaultValue } = props;
  return (
    <div className={styles.field}>
      <label>{name}</label>
      <input name={name} defaultValue={defaultValue} ref={ref} />
    </div>
  );
});

Input.displayName = "Input";

export default Input;
