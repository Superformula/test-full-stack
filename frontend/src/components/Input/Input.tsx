import React from 'react';
import styles from './Input.module.scss';

type Props = {
  id: string;
  label?: string;
  placeholder?: string;
  value?: string;
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
};

const Input: React.FC<Props> = ({
  id,
  label,
  placeholder,
  value,
  onChange = () => {
    /**/
  },
}) => (
  <div className={styles.wrapper}>
    {label && (
      <label htmlFor={id} className={styles.label}>
        {label}
      </label>
    )}
    <input
      id={id}
      className={styles.input}
      placeholder={placeholder}
      value={value || ''}
      onChange={onChange}
    />
  </div>
);

export default Input;
