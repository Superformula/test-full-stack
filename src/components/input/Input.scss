@import "src/styles/variables";

.input {
  &-wrapper {
    padding: 0;
    display: inline-block;
    width: 400px;
    max-width: 100%;

    label {
      display: block;
      font-weight: 600;
      font-size: 18px;
      line-height: 23px;
      color: $color-black;
      margin-bottom: 8px;
    }
  }

  &-field {
    width: 100%;
    height: 64px;

    background: $color-white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    border-radius: $border-radius;
    font-family: Source Sans Pro, serif;

    padding: 16px;
    box-sizing: border-box;

    font-size: 24px;
    line-height: 30px;
    color: $color-black;

    position: relative;

    &:focus {
      outline: none;
      border: 1px solid rgba(0, 0, 0, 0.5);
    }
  }
}


