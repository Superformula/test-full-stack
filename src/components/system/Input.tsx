import styled from "@emotion/styled";

export const Input = styled.input`
  align-items: center;
  background-color: ${({ theme }) => theme.color.bgPrimary};
  border-radius: 8px;
  border: 1px solid ${({ theme }) => theme.color.divider};
  color: ${({ theme }) => theme.color.primary};
  display: flex;
  font-family: Source Sans Pro;
  font-size: 1.5rem;
  font-weight: 400;
  line-height: 1.875rem;
  padding: 1rem;
  transition: border 0.3s ease-in-out;
  width: 25rem;

  &:hover {
    border-color: ${({ theme }) => theme.color.hover};
  }
  &:focus {
    outline: none;
    border-color: ${({ theme }) => theme.color.focus};
  }
  ::placeholder {
    color: ${({ theme }) => theme.color.secondary};
    font-weight: 300;
  }
`;
