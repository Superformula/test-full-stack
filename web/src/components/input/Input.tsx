import classNames from 'classnames'
import React, { ReactElement, useMemo } from 'react'

import style from './Input.module.scss'

type InputProps = React.HTMLProps<HTMLInputElement> & {
  /**
   * This variable, when set to false, won't create any padding on top
   * of the field and nor any placeholder will be present when the field
   * has value
   */
  placeholderOnTop?: boolean
}

export const Input = ({ placeholderOnTop, ...props }: InputProps): ReactElement => {
  const classes = useMemo(
    () =>
      classNames({
        [style.input]: true,
        [style.padded]: placeholderOnTop !== false,
      }),
    [placeholderOnTop],
  )
  return (
    <div className={classes}>
      {placeholderOnTop !== false && props.placeholder && props.value && (
        <div className={style.placeholder}>{props.placeholder}</div>
      )}
      <input {...props} />
    </div>
  )
}
