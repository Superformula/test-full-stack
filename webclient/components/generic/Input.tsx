import { ReactElement, InputHTMLAttributes } from 'react'

import { HTMLElementType } from '../../interfaces'

import styles from './Input.module.css'

interface HTMLInputProps
  extends InputHTMLAttributes<HTMLInputElement | HTMLTextAreaElement> {
  elementType?: HTMLElementType;
  rows?: number;
}

export default function Input({
  elementType = 'input',
  className,
  ...props
}: HTMLInputProps): ReactElement {
  const InputElement = elementType

  return (
    <InputElement {...props} className={`${styles.textbox} ${className}`} />
  )
}
