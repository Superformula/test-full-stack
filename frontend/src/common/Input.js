import React from 'react'
import './Input.css'

export const Input = ({ value, onChange, label }) => {
  const handleOnChange = (evt) => {
    const { value } = evt.target
    onChange(value)
  }
  return (
    <div className={'input-container'}>
      <p className={'input-label'}>{label}</p>
      <input className={'input-text-box'} value={value} onChange={handleOnChange} placeholder={'Type...'} />
    </div>
  )
}
