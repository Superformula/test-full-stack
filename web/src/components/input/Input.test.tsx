import { render } from '@testing-library/react'
import React from 'react'
import { Input } from './Input'

test('renders input and matches snapshot', () => {
  const { container } = render(<Input placeholder="Hello World" />)
  expect(container).toMatchSnapshot()
})

test('renders input with value and matches snapshot', () => {
  const onChange = (evt: React.ChangeEvent<HTMLInputElement>) => {}
  const { container, getByText } = render(<Input placeholder="Hello" value="World" onChange={onChange} />)
  const placeholderElement = getByText(/Hello/i)
  expect(container).toMatchSnapshot()
  expect(placeholderElement).toBeInTheDocument()
})
