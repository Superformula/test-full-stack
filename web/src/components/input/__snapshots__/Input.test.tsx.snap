// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders input and matches snapshot 1`] = `
<div>
  <div
    class="input"
  >
    <input
      placeholder="Hello World"
    />
  </div>
</div>
`;

exports[`renders input with value and matches snapshot 1`] = `
<div>
  <div
    class="input"
  >
    <div
      class="placeholder"
    >
      Hello
    </div>
    <input
      placeholder="Hello"
      value="World"
    />
  </div>
</div>
`;
